File: api/audio-book/src/main/java/org/backend/user/dto/request/UpdateUserRequest.java
package org.backend.user.dto.request;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;
import org.backend.user.enums.RoleEnum;

@Getter
@Setter
public class UpdateUserRequest {
    @NotBlank(message = "Name is required")
    private String name;

    @Email(message = "Email is not valid")
    @NotBlank(message = "Email is required")
    private String email;

    private String password;

    @NotNull(message = "Avatar file id is required")
    private Long avatarFileId;

    private RoleEnum role;
}

File: mobile_client/lib/src/admin/user/models/update_user_request.dart
import 'user_response.dart';

class UpdateUserRequest {
  final String name;
  final String email;
  final String? password;
  final int avatarFileId;
  final RoleEnum role;

  UpdateUserRequest({
    required this.name,
    required this.email,
    this.password,
    required this.avatarFileId,
    required this.role,
  });

  Map<String, dynamic> toJson() {
    final data = <String, dynamic>{
      'name': name,
      'email': email,
      'avatarFileId': avatarFileId,
      'role': role.toBackendValue(),
    };

    if (password != null && password!.trim().isNotEmpty) {
      data['password'] = password;
    }

    return data;
  }
}